An app using Roboguice and Proguard crashes at launch

Here is the stacktrace:

```
E/AndroidRuntime( 7844): FATAL EXCEPTION: main
E/AndroidRuntime( 7844): java.lang.RuntimeException: Unable to start activity ComponentInfo{com.androidkickstartr.app/com.androidkickstartr.app.MainActivity}: com.google.inject.CreationException: Guice creation errors:
E/AndroidRuntime( 7844):
E/AndroidRuntime( 7844): 1) An exception was caught and reported. Message: Missing type parameter.
E/AndroidRuntime( 7844):   at com.google.inject.internal.InjectorShell$Builder.build(SourceFile:133)
E/AndroidRuntime( 7844):
E/AndroidRuntime( 7844): 2) No implementation for android.content.Context was bound.
E/AndroidRuntime( 7844):   at roboguice.config.DefaultRoboModule.configure(SourceFile:130)
E/AndroidRuntime( 7844):
E/AndroidRuntime( 7844): 2 errors
E/AndroidRuntime( 7844):    at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2180)
E/AndroidRuntime( 7844):    at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2230)
E/AndroidRuntime( 7844):    at android.app.ActivityThread.access$600(ActivityThread.java:141)
E/AndroidRuntime( 7844):    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1234)
E/AndroidRuntime( 7844):    at android.os.Handler.dispatchMessage(Handler.java:99)
E/AndroidRuntime( 7844):    at android.os.Looper.loop(Looper.java:137)
E/AndroidRuntime( 7844):    at android.app.ActivityThread.main(ActivityThread.java:5041)
E/AndroidRuntime( 7844):    at java.lang.reflect.Method.invokeNative(Native Method)
E/AndroidRuntime( 7844):    at java.lang.reflect.Method.invoke(Method.java:511)
E/AndroidRuntime( 7844):    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:793)
E/AndroidRuntime( 7844):    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:560)
E/AndroidRuntime( 7844):    at dalvik.system.NativeStart.main(Native Method)
E/AndroidRuntime( 7844): Caused by: com.google.inject.CreationException: Guice creation errors:
E/AndroidRuntime( 7844):
E/AndroidRuntime( 7844): 1) An exception was caught and reported. Message: Missing type parameter.
E/AndroidRuntime( 7844):   at com.google.inject.internal.InjectorShell$Builder.build(SourceFile:133)
E/AndroidRuntime( 7844):
E/AndroidRuntime( 7844): 2) No implementation for android.content.Context was bound.
E/AndroidRuntime( 7844):   at roboguice.config.DefaultRoboModule.configure(SourceFile:130)
E/AndroidRuntime( 7844):
E/AndroidRuntime( 7844): 2 errors
E/AndroidRuntime( 7844):    at com.google.inject.internal.Errors.throwCreationExceptionIfErrorsExist(SourceFile:435)
E/AndroidRuntime( 7844):    at com.google.inject.internal.InternalInjectorCreator.initializeStatically(SourceFile:154)
E/AndroidRuntime( 7844):    at com.google.inject.internal.InternalInjectorCreator.build(SourceFile:106)
E/AndroidRuntime( 7844):    at com.google.inject.Guice.createInjector(SourceFile:95)
E/AndroidRuntime( 7844):    at com.google.inject.Guice.createInjector(SourceFile:83)
E/AndroidRuntime( 7844):    at roboguice.RoboGuice.setBaseApplicationInjector(SourceFile:94)
E/AndroidRuntime( 7844):    at roboguice.RoboGuice.setBaseApplicationInjector(SourceFile:139)
E/AndroidRuntime( 7844):    at roboguice.RoboGuice.getBaseApplicationInjector(SourceFile:59)
E/AndroidRuntime( 7844):    at roboguice.RoboGuice.getInjector(SourceFile:149)
E/AndroidRuntime( 7844):    at com.androidkickstartr.app.robosherlock.RoboSherlockFragmentActivity.onCreate(SourceFile:31)
E/AndroidRuntime( 7844):    at com.androidkickstartr.app.MainActivity.onCreate(SourceFile:15)
E/AndroidRuntime( 7844):    at android.app.Activity.performCreate(Activity.java:5104)
E/AndroidRuntime( 7844):    at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1080)
E/AndroidRuntime( 7844):    at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2144)
E/AndroidRuntime( 7844):    ... 11 more
E/AndroidRuntime( 7844): Caused by: java.lang.RuntimeException: Missing type parameter.
E/AndroidRuntime( 7844):    at com.google.inject.TypeLiteral.getSuperclassTypeParameter(SourceFile:99)
E/AndroidRuntime( 7844):    at com.google.inject.TypeLiteral.<init>(SourceFile:77)
E/AndroidRuntime( 7844):    at roboguice.config.DefaultRoboModule$1.<init>(SourceFile:130)
E/AndroidRuntime( 7844):    at roboguice.config.DefaultRoboModule.configure(SourceFile:130)
E/AndroidRuntime( 7844):    at com.google.inject.AbstractModule.configure(SourceFile:59)
E/AndroidRuntime( 7844):    at com.google.inject.spi.Elements$RecordingBinder.install(SourceFile:223)
E/AndroidRuntime( 7844):    at com.google.inject.spi.Elements.getElements(SourceFile:101)
E/AndroidRuntime( 7844):    at com.google.inject.internal.InjectorShell$Builder.build(SourceFile:133)
E/AndroidRuntime( 7844):    at com.google.inject.internal.InternalInjectorCreator.build(SourceFile:103)
E/AndroidRuntime( 7844):    ... 22 more
```

--- a/androidkickstartr-core/src/main/resources/templates/Proguard.ftl
+++ b/androidkickstartr-core/src/main/resources/templates/Proguard.ftl
@@ -1,8 +1,9 @@
+-optimizationpasses 5
 -dontusemixedcaseclassnames
 -dontskipnonpubliclibraryclasses
 -dontpreverify
 -verbose
--optimizations !code/simplification/arithmetic,!field/*,!class/merging/*
+-optimizations !code/simplification/arithmetic,!field/*,!class/merging/*,!code/allocation/variable
 -ignorewarnings
 -renamesourcefileattribute SourceFile
 -keepattributes SourceFile,LineNumberTable,*Annotation*,Signature

--- a/androidkickstartr-core/src/main/resources/templates/Proguard.ftl
+++ b/androidkickstartr-core/src/main/resources/templates/Proguard.ftl
@@ -1,4 +1,3 @@
--optimizationpasses 5
 -dontusemixedcaseclassnames
 -dontskipnonpubliclibraryclasses
 -dontpreverify

--- a/androidkickstartr-core/src/main/resources/templates/Proguard.ftl
+++ b/androidkickstartr-core/src/main/resources/templates/Proguard.ftl
@@ -1,6 +1,13 @@
+-optimizationpasses 5
+-dontusemixedcaseclassnames
+-dontskipnonpubliclibraryclasses
 -dontpreverify
--optimizations !code/simplification/arithmetic,!field/*,!class/merging/*
 -verbose
+-optimizations !code/simplification/arithmetic,!field/*,!class/merging/*
+-ignorewarnings
+-renamesourcefileattribute SourceFile
+-keepattributes SourceFile,LineNumberTable,*Annotation*,Signature
+ 
 
 -keepattributes *Annotation*
 
@@ -116,15 +123,15 @@
     @com.google.inject.Inject <init>(...);
 }
 
-# from https://groups.google.com/forum/?fromgroups=#!topic/roboguice/vcA1Z0biPqI
--keep class com.google.inject.** { *; } 
+-keep class com.google.inject.** { *; }
+-keep class javax.inject.** { *; }
+-keep class javax.annotation.** { *; }
+-keep class roboguice.** { *; }
 
 # There's no way to keep all @Observes methods, so use the On*Event convention to identify event handlers
 -keepclassmembers class * { 
     void *(**On*Event);
 }
-
--keep public class roboguice.**
 
 -dontwarn roboguice.**
 

